File: examples/nodejs/detect_empty_backup_file.js
#!/usr/bin/env node
/**
 * Detect Empty Backup File
 * Problem: Backup file is created but is zero bytes
 * Solution: Send file size in payload, validate in DeadManPing panel
 * Documentation: https://deadmanping.com/detect-empty-backup-file
 */

const fs = require('fs');
const { execSync } = require('child_process');
const https = require('https');

const MONITOR_ID = 'YOUR_MONITOR_ID';  // Replace with your actual monitor ID
const backupFile = '/backups/db-backup.sql.gz';

// Run backup
execSync(`pg_dump mydb | gzip > ${backupFile}`);

// Get file size (0 if file doesn't exist)
let fileSize = 0;
try {
  fileSize = fs.statSync(backupFile).size;
} catch (e) {
  // File doesn't exist
}

// Always ping with file size in payload
// In DeadManPing panel: set validation rules:
//   - "size" > 0 (to detect empty files)
//   - "size" >= 1024 (minimum size, optional)
// Panel will automatically detect if size violates rules
https.request(`https://deadmanping.com/api/ping/${MONITOR_ID}?size=${fileSize}`, { method: 'POST' }).end();

File: examples/python/detect_empty_backup_file.py
#!/usr/bin/env python3
"""
Detect Empty Backup File
Problem: Backup file is created but is zero bytes
Solution: Send file size in payload, validate in DeadManPing panel
Documentation: https://deadmanping.com/detect-empty-backup-file
"""

import os
import subprocess
import requests

MONITOR_ID = "YOUR_MONITOR_ID"  # Replace with your actual monitor ID
backup_file = "/backups/db-backup.sql.gz"

# Run backup
subprocess.run(["pg_dump", "mydb"], stdout=open(backup_file.replace('.gz', ''), "w"))
subprocess.run(["gzip", backup_file.replace('.gz', '')])

# Get file size (0 if file doesn't exist)
file_size = os.path.getsize(backup_file) if os.path.exists(backup_file) else 0

# Always ping with file size in payload
# In DeadManPing panel: set validation rules:
#   - "size" > 0 (to detect empty files)
#   - "size" >= 1024 (minimum size, optional)
# Panel will automatically detect if size violates rules
requests.post(f"https://deadmanping.com/api/ping/{MONITOR_ID}?size={file_size}")
